osquery/tables/applications/vscode_extensions.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <set>
+#include <string>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/tables/system/system_utils.h>
+#include <osquery/utils/json/json.h>
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+void genReadJSONAndAddExtensionRows(const std::string& uid,
+                                    const std::string& path,
+                                    QueryData& results) {
+  std::string json;
+  if (!readFile(path, json).ok()) {
+    LOG(WARNING) << "Could not read vscode extensions.json from '" << path
+                 << "'";
+    return;
+  }
+
+  auto doc = JSON::newArray();
+  if (!doc.fromString(json) || !doc.doc().IsArray()) {
+    LOG(WARNING) << "Could not parse vscode extensions.json from " << path
+                 << "'";
+    return;
+  }
+
+  rapidjson::Value::ConstValueIterator itr;
+  for (itr = doc.doc().Begin(); itr != doc.doc().End(); ++itr) {
+    const rapidjson::Value& extension = *itr;
+    const rapidjson::Value& identifier = extension["identifier"];
```

Comment:
Careful that accessing `extension` like that presumes it's an object, while it's not guaranteed here; check if `IsObject` first.